Utilities/XrdAdaptor/plugins/XrdStorageMaker.cc

```diff
@@ -83,7 +83,16 @@ class XrdStorageMaker final : public StorageMaker
     XrdCl::URL url(fullpath);
     XrdCl::FileSystem fs(url);
     std::vector<std::string> fileList; fileList.push_back(url.GetPath());
-    fs.Prepare(fileList, XrdCl::PrepareFlags::Stage, 0, &m_null_handler);
+    XrdCl::XRootDStatus status;
+    status = fs.Prepare(fileList, XrdCl::PrepareFlags::Stage, 0, &m_null_handler);
+    if (!status.IsOK())
+    {
+        cms::Exception ex("StageInError");
+        ex << "XrdCl::FileSystem::Prepare failed with error '"
+           << status.ToStr() << "' (errNo = " << status.errNo << ")";
+        ex.addContext("Calling XrdStorageMaker::stagein");
+        throw ex;
+    }
```

Comment:
Hm - the `prepare` is advisory (although it would be quite strange to fail here).  Can you downgrade this to a warning message?